### DISCUSSION TITLE
I am getting Method GET not allowed! while signing.

### DISCUSSION THREAD

**Discussion by ats1999** [nextauthjs/next-auth#1004] [Help]
## I am getting `Method GET not allowed!` while signing.
**Code:**
_**[...nextauth].js**_
```js
import NextAuth from 'next-auth'
import Providers from 'next-auth/providers'

const options = {
  // Configure one or more authentication providers
  providers: [
    Providers.Google({
      clientId: process.env.GOOGLE_ID,
      clientSecret: process.env.GOOGLE_SECRET,
      // authorizationUrl: 'https://accounts.google.com/o/oauth2/v2/auth?prompt=consent&access_type=offline&response_type=code',
    }),
  ],
  // A database is optional, but required to persist accounts in a database
  database: process.env.MONGO_URI,
}

export default (req, res) => NextAuth(req, res, options)
```
**page component**
```js
import React from 'react'
import { signIn, signOut, useSession } from 'next-auth/client'

export default function Page() {
  const [ session, loading ] = useSession()

  return <>
    {!session && <>
      Not signed in <br/>
      <button onClick={signIn("google")}>Sign in</button>
    </>}
    {session && <>
      Signed in as {session.user.email} <br/>
      <button onClick={signOut}>Sign out</button>
    </>}
  </>
}
```

This code displays me a signin button, when i clicked, then it takes me on this url.
```js
http://localhost:3000/api/auth/signin?callbackUrl=http%3A%2F%2Flocalhost%3A3000%2Ftest
```
I got this on the browser screen.
```
Method GET not allowed!
```
This is logged on the console.
```js
GET http://localhost:3000/api/auth/signin?callbackUrl=http%3A%2F%2Flocalhost%3A3000%2Ftest 405 (Method Not Allowed)
```

**Comment by djt** [comment]
Did you manage to resolve this?  I'm having the same issue.

**Reply by djt** [reply]
For anyone else having this issue, I was able to resolve it by deleting and recreating my [...nextauth].js file.

**Comment by Koala-afurubayashi** [comment]
I still have a similar problem. It occurs in both V4 and V3. Isn't the above rebuilt or cured by updating the server?

**Comment by junaid-ahmad** [comment]
I was also getting 405 Error when trying to enable Google Provider for NEXT Auth in Next.js 13 and 14. Solution is very simple, just delete [...nextauth] folder and recreate it.

**Comment by RodrigoBLISS** [comment]
Looks like NextAuth has an issue with NextJS14 and SRC project instead of Pages.
Deleting and recreating […nextauth] did not work, and probably I would be worried if it did…

What happened in my case was that I had to encapsulate the NextAuth into the proper requests.

-> Step 1 :
[…nextauth] /route.js

const myNextAuthOptions = {} // Your options here…

export async function GET( req,res ) {
return NextAuth(req,res,MyNextAuthOption)
}

export async function POST( req,res ) {
return NextAuth(req,res,MyNextAuthOption)
}

[!] New Error :
Yay, you are almost there. In my case NextJS gave me a different error, complaining that
my ./core/routes were not exported on the package.json from NextAuth…

-> Step 2 :
edit node_modules/next-auth/package.json
Add :
“./core/routes”: {
“types”: “./core/routes/.d.ts",
“default”: "./core/routes/.js”
}
to “exports”

This fixed for me. Hope it could help you.

**Comment by dikaaajs** [comment]
just add this code

`
const handler = NextAuth(authOptions);

export { handler as GET, handler as POST };
`

**Reply by Andrew213** [reply]
where

**Reply by dikaaajs** [reply]
you can see it in my repo https://github.com/dikaaajs/qroom/tree/master/app/api/auth/%5B...nextauth%5D

**Reply by Francopf21** [reply]
this fix mi problem, ty

**Reply by MiracleForge** [reply]
Thank you @dikaaajs , this help me a lot

**Reply by iscjorgemena** [reply]
Thanks

**Comment by Faraz1995** [comment]
i still have the same error
here is my app/api/auth/[...nextauth]/route.ts file
`import nextAuth from 'next-auth'
import GoogleProvider from 'next-auth/providers/google'

const handler = nextAuth({
  providers: [
    GoogleProvider({
      clientId: process.env.GOOGLE_CLIENT_ID as string,
      clientSecret: process.env.GOOGLE_CLIENT_SECRET as string,
      authorization: {
        params: {
          prompt: "consent",
          access_type: "offline",
          response_type: "code"
        }
      }
    })
  ]
})

export { handler as GET, handler as Post }`
what should i do?

**Reply by dikaaajs** [reply]
```
export const authOptions: NextAuthOptions = {
  session: {
    strategy: "jwt",
  },
  providers: [
    Credentials({
      name: "credentials",
      credentials: {},
      async authorize(credentials: any, req) {
        const { username, password } = credentials as {
          username: string;
          password: string;
        };

        // login
        await connectMongoDB();
        let findUser = await Account.findOne({ username });
        if (findUser === null) {
          return null;
        }

        const passwordMatch = await bcrypt.compare(password, findUser.password);
        if (!passwordMatch) {
          return null;
        }

        const tmp = findUser;
        findUser.name = tmp.username;
        return findUser;
      },
    }),
  ],
  pages: {
    signIn: "/auth/login",
  },
  callbacks: {
    async session({ session, user }) {
      return session;
    },
  },
  secret: process.env.NEXTAUTH_SECRET,
};

const handler = NextAuth(authOptions);
export { handler as GET, handler as POST };
```

try copy paste this code
